libr/asm/p/asm_x86_cs.c

```diff
@@ -147,10 +147,7 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 				insn->mnemonic, insn->op_str[0]?" ":"",
 				insn->op_str);
 		if (a->config->syntax != R_ASM_SYNTAX_MASM) {
-			char *ptrstr = strstr (buf_asm, "ptr ");
-			if (ptrstr) {
-				memmove (ptrstr, ptrstr + 4, strlen (ptrstr + 4) + 1);
-			}
+			r_str_replace (buf_asm, "ptr ", "", true);
```

Comment:
The return pointer can differ from buf asm . Take care of it

Reply:
Wasnt there a failing test btw?

Reply:
only if strlen(key) < strlen(val), i checked it